### PR TITLE
fix: get google apt key with gpg commmand

### PR DESCRIPTION
**Pull request by volatilemolotov**
Changes the way google apt key was ingested. It now uses `gpg` command instead of `tee` which is the way google official documentation does it. 
Something changed in the way apt handles keys

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

File: playground/infrastructure/cloudbuild/cloudbuild_playground_infra.sh [hunk 1]
```diff
@@ -40,16 +40,31 @@ apt-get -qq install -y google-cloud-sdk-gke-gcloud-auth-plugin google-cloud-sdk
...
+                else
+                        echo "Cannot create bucket, gcloud exit code: $?"
+                        exit 1
+                fi
+elif [ $checkBucket -eq 401 ]; then
+        echo "You do not have permission to access this bucket"
+        exit 1
 else
-        echo "Cannot create bucket, please check if the name is unique"
+        echo "Something went wrong, please check the bucket name"
```
Comment (Abacn) [inline playground/infrastructure/cloudbuild/cloudbuild_playground_infra.sh:62]: Can we have a more specific error message?

**Review by Abacn — APPROVED** [review]
Thanks! Just had a few minor comments

File: playground/infrastructure/cloudbuild/cloudbuild_playground_infra.sh (hunk 1 above)
Comment (volatilemolotov) [reply playground/infrastructure/cloudbuild/cloudbuild_playground_infra.sh]: Its part of the else wiring. Added a write of the error code, not sure what else to expect

File: playground/infrastructure/cloudbuild/cloudbuild_playground_infra.sh [hunk 2]
```diff
@@ -40,16 +40,31 @@ apt-get -qq install -y google-cloud-sdk-gke-gcloud-auth-plugin google-cloud-sdk
...
 cd beam
 mkdir playground/terraform/environment/$ENVIRONMENT_NAME
- 
-gcloud storage buckets create gs://$TF_VAR_state_bucket --location $TF_VAR_region --project $TF_VAR_project_id --pap
-if [ $? -eq 0 ]
-then 
-        echo "Creation successfull"
+
+
+checkBucket=$(curl -s -H "X-Goog-User-Project: $(gcloud config get-value project)" -H "Authorization: Bearer $(gcloud auth print-access-token)" https://storage.googleapis.com/storage/v1/b/$TF_VAR_state_bucket -w "%{http_code}" -o /dev/null)
```
[the remark this answers is not in the document]
Comment (volatilemolotov) [reply playground/infrastructure/cloudbuild/cloudbuild_playground_infra.sh:46]: Added comment explaning that it checks for bucket, creates if missing

**Pull request merged**